src/array/cpu/coo_remove.cc

```diff
@@ -0,0 +1,48 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file array/cpu/coo_remove.cc
+ * \brief COO matrix remove entries CPU implementation
+ */
+#include <dgl/array.h>
+#include <utility>
+#include <vector>
+#include "array_utils.h"
+
+namespace dgl {
+using runtime::NDArray;
+namespace aten {
+namespace impl {
+
+template <DLDeviceType XPU, typename IdType>
+std::pair<COOMatrix, IdArray> COORemove(COOMatrix coo, IdArray entries) {
+  const int64_t nnz = coo.row->shape[0];
+  const IdType *row_data = static_cast<IdType *>(coo.row->data);
+  const IdType *col_data = static_cast<IdType *>(coo.col->data);
+  const IdType *eid_data = COOHasData(coo) ? static_cast<IdType *>(coo.data->data) : nullptr;
+
+  IdHashMap<IdType> eid_map(entries);
+
+  std::vector<IdType> new_rows, new_cols, new_eids;
```

Comment:
`new_rows.reserve(nnz - entries->shape[0])`. It is the most common case and also a lower bound. Same for `new_cols` and `new_eids`.